src/core/instance/instance.hpp

```diff
@@ -631,7 +631,9 @@ class Instance : public otInstance, private NonCopyable
     Utils::ChannelMonitor mChannelMonitor;
 #endif
 
-#if OPENTHREAD_CONFIG_CHANNEL_MANAGER_ENABLE && OPENTHREAD_FTD
+#if (OPENTHREAD_CONFIG_CHANNEL_MANAGER_ENABLE && OPENTHREAD_FTD) ||                   \
+    (OPENTHREAD_CONFIG_CHANNEL_MANAGER_CSL_CHANNEL_SELECT_ENABLE && OPENTHREAD_MTD && \
+     OPENTHREAD_CONFIG_MAC_CSL_RECEIVER_ENABLE)
```

Comment:
I think we can simplify the checks. 

How about the following:
```c++
#if OPENTHREAD_CONFIG_CHANNEL_MANAGER_ENABLE && (OPENTHREAD_FTD || OPENTHREAD_CONFIG_CHANNEL_MANAGER_CSL_CHANNEL_SELECT_ENABLE)
```
We say we should have "channel manager" feature itself enabled, then on FTD it does it channel change process,  otherwise we need the `CSL_CHANNEL_SELECT` to be enabled.

----

We can also add a check at the top of `channel_manager.hpp` to make sure the CSL related configs are enabled together. This way we can simplify all the `#if` checks. We can add something like this:
```c++
#if OPENTHREAD_CONFIG_CHANNEL_MANAGER_CSL_CHANNEL_SELECT_ENABLE && !OPENTHREAD_CONFIG_MAC_CSL_RECEIVER_ENABLE
#error "OPENTHREAD_CONFIG_CHANNEL_MANAGER_CSL_CHANNEL_SELECT_ENABLE requires OPENTHREAD_CONFIG_MAC_CSL_RECEIVER_ENABLE"
#endif
```

Reply:
Ok. I am not fully aware of the consequences for builds other than FTD and MTD. However, I changed the checks according your proposal.

Reply:
Thanks.